### PR TITLE
Subgraph library improvements

### PR DESCRIPTION
**Pull request by saleel**
## Description

- Add query filters to subgraph library
- Add option to specify subgraph URL in subgraph library

## Related Issue

#242 
#243 

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: packages/subgraph/src/subgraph.ts [hunk 1]
```diff
@@ -30,19 +55,40 @@ export default class Subgraph {
      * @param options Options to select the group parameters.
      * @returns List of groups.
      */
-    async getGroups(options: GroupOptions = {}): Promise<any[]> {
+    async getGroups(options: GroupOptions = {}): Promise<GetGroupsResponse[]> {
```
Comment (cedoor) [inline packages/subgraph/src/subgraph.ts]: Should we add the `GetGroupsResponse` return type to the `getGroup` function too?

File: packages/subgraph/src/subgraph.ts [hunk 2]
```diff
@@ -3,6 +3,26 @@ import checkParameter from "./checkParameter"
...
+    }
+    admin: string
+    members: string[]
+    verifiedProofs: {
+        signal: string
+        merkleTreeRoot: string
+        externalNullifier: string
+        nullifierHash: string
+        timestamp: string
+    }[]
```
Comment (cedoor) [inline packages/subgraph/src/subgraph.ts]:
```suggestion
    members?: string[]
    verifiedProofs?: {
        signal: string
        merkleTreeRoot: string
        externalNullifier: string
        nullifierHash: string
        timestamp: string
    }[]
```

**Review by cedoor — CHANGES_REQUESTED** [review]

File: packages/subgraph/src/subgraph.ts (hunk 1 above)
Comment (saleel) [reply packages/subgraph/src/subgraph.ts]: Ah yes, missed that. Added now.

File: packages/subgraph/src/subgraph.ts (hunk 2 above)
Comment (saleel) [reply packages/subgraph/src/subgraph.ts]: I thought of adding Partial<> to the response, but then consuming apps would need to typecast as not null. 
I have added this now anyway.

**Review by cedoor — APPROVED** [review]

**Pull request merged**